src/main/java/com/redislabs/redisai/Tensor.java

```diff
@@ -1,16 +1,73 @@
 package com.redislabs.redisai;
 
+import com.redislabs.redisai.exceptions.JRedisAIRunTimeException;
+import java.util.ArrayList;
+import java.util.List;
+import redis.clients.jedis.Protocol;
+import redis.clients.jedis.util.SafeEncoder;
+
 public class Tensor {
   private DataType dataType;
   private long[] shape;
   private Object values;
 
+  /**
+   * @param dataType
+   * @param shape
+   * @param values
+   */
   public Tensor(DataType dataType, long[] shape, Object values) {
     this.shape = shape;
     this.values = values;
     this.dataType = dataType;
   }
 
+  /**
+   * Given a RESP reply from RedisAI this method will create a new Tensor
+   *
+   * @param reply reply from RedisAI Server
+   * @return Tensor Object
+   */
+  protected static Tensor createTensorFromRespReply(List<?> reply) {
+    DataType dtype = null;
+    long[] shape = null;
+    Object values = null;
+    Tensor tensor = null;
+    for (int i = 0; i < reply.size(); i += 2) {
+      String arrayKey = SafeEncoder.encode((byte[]) reply.get(i));
+      switch (arrayKey) {
+        case "dtype":
+          String dtypeString = SafeEncoder.encode((byte[]) reply.get(i + 1));
+          dtype = DataType.getDataTypefromString(dtypeString);
+          if (dtype == null) {
+            throw new JRedisAIRunTimeException("Unrecognized datatype: " + dtypeString);
+          }
+          break;
+        case "shape":
+          List<Long> shapeResp = (List<Long>) reply.get(i + 1);
+          shape = new long[shapeResp.size()];
+          for (int j = 0; j < shapeResp.size(); j++) {
+            shape[j] = shapeResp.get(j);
+          }
+          break;
+        case "values":
+          if (dtype == null) {
+            throw new JRedisAIRunTimeException(
+                "Trying to decode values array without previous datatype info");
+          }
+          List<byte[]> valuesEncoded = (List<byte[]>) reply.get(i + 1);
+          values = dtype.toObject(valuesEncoded);
+          break;
+        default:
+          break;
+      }
+    }
+    if (dtype != null && shape != null && values != null) {
+      tensor = new Tensor(dtype, shape, values);
+    }
+    return tensor;
```

Comment:
doesn't that just return null?
Shouldn't we throw an error?

Reply:
Yes and agree. making the changes to throw an error

Reply:
doing the same for MODELGET and SCRIPTGET replies.

Reply:
solved on the latest commit